Backporting latest Aktualizr to branch Morty

leon-anavi opened the pull request:
Hi,

These git commits backport latest stable Aktualizr and OSTree to branch Morty of meta-updater. The version of Aktualizr is the same as the one in branches Sumo and Thrud. OSTree has been upgraded to  v2018.7. Additional recipes for dependencies have been added, including asn1c and bbappend for building libsodium-native. All changes have been tested with end to end updates on QEMUx86-64 and Raspberry Pi.

Best regards, Leon

patrickvacek commented:
At first glance, looks promising! Thanks for putting this together. I'm testing now, but I see two problems so far:

1. `.pyc` files shouldn't be committed.
2. We will also need to update oe-selftest. It looks like we only started using it in pyro, but the standard way of testing meta-updater is with that, so we should backport that as well. I would try starting with the rocko version; the pyro version is out of date and sumo/thud have changed enough that it might be a bit of work to deal with them. The rocko version should be functionally on par with sumo/thud/master, though.

leon-anavi commented:
Hi @patrickvacek,

Thank you for the feedback. Based on your recommendations I have just rebased git commit 92acec81196e731edc9a3b3e170e56967215a21b "scripts: Backporting scripts for QEMU". The pyc file is no longer included and I have added *.pyc to the .gitignore.

I am on a business trip and I have limited Internet access. I will look into the tests but most probably I will be able to dedicate more time and focus on their backporting next week. In the mean time if you have no other remarks could you merge this GitHub PR and after that I will make a separate PR for the tests?

Thanks,
Leon

patrickvacek commented:
> The pyc file is no longer included and I have added *.pyc to the .gitignore.

Thanks!

> In the mean time if you have no other remarks could you merge this GitHub PR and after that I will make a separate PR for the tests?

I'd prefer to include the tests before merging this, otherwise it's hard to verify that everything really is right. That said, I've built qemu and rpi images without a problem a did a full roundtrip with an rpi, so we're definitely on the right track. I haven't done a full code review yet, though.

patrickvacek commented on recipes-sota/aktualizr/aktualizr-ca-implicit-prov-creds.bb:
```diff
@@ -0,0 +1,51 @@
+        install -m 0700 -d ${D}${localstatedir}/sota
+        aktualizr_cert_provider --credentials ${SOTA_PACKED_CREDENTIALS} \
+                                --device-ca ${SOTA_CACERT_PATH} \
+                                --device-ca-key ${SOTA_CAKEY_PATH} \
```
`aktualizr_cert_provider` has been renamed to `aktualizr-cert-provider`, and `--device-ca` and `--device-ca-key` are now `--fleet-ca` and `--fleet-ca-key`. This is the type of thing oe-selftest can help with catching!

patrickvacek commented on classes/sota.bbclass:
```diff
@@ -5,29 +5,47 @@ python __anonymous() {
-SOTA_MACHINE_raspberrypi3 ?= "raspberrypi"
+SOTA_MACHINE_rpi ?= "raspberrypi"
 SOTA_MACHINE_porter ?= "porter"
 SOTA_MACHINE_m3ulcb = "m3ulcb"
```
We might as well add the h3ulcb line here as well to be consistent with other branches.

patrickvacek commented on recipes-sota/aktualizr/aktualizr_git.bb:
```diff
@@ -4,42 +4,133 @@ HOMEPAGE = "https://github.com/advancedtelematic/aktualizr"
+    install -m 0644 ${WORKDIR}/aktualizr-secondary.service ${D}${systemd_unitdir}/system/aktualizr-secondary.service
+    install -m 0700 -d ${D}${libdir}/sota/conf.d
+    install -m 0700 -d ${D}${sysconfdir}/sota/conf.d
+
```
There's a brief bit of logic for handling a `SOTA_HARDWARE_ID` that is missing here.

patrickvacek commented on recipes-sota/ostree/ostree_git.bb:
```diff
@@ -73,17 +76,24 @@ do_install_append_class-native() {
 DEPENDS_remove_class-native = "systemd-native"
-
+FILES_${PN}-dev += " ${datadir}/gir-1.0"
+FILES_${PN}-dracut = "${sysconfdir}/dracut.conf.d ${libdir}/dracut"
```
There was a bunch of cleanup done of the ostree recipe over the past few months. It looks like a lot of it is here, but I still see substantial differences between the rocko version and this one. Is that necessary due to limitations in morty, or can we get them closer in sync?

patrickvacek commented:
Other files missing from the backport:
1. `classes/target_version_example.bbclass`
2. `recipes-sota/config/`
3. `recipes-support/libp11`, `recipes-support/slcand-start`, `recipes-support/softhsm/`, and `recipes-support/softhsm-testtoken/`
4. `recipes-test/demo-network-config/` and `recipes-test/images/`

All of those can probably be copied verbatim from rocko.

Other files that should be updated with changes from later branches:
1. `CONTRIBUTING.adoc`
2. `README.adoc`

Those should probably also simply be identical to the rocko versions.

`recipes-sota/sota-tools/` should be removed. (`rvi-sota-client/` could be as well, and thus the whole meta-rust layer as well, but you don't need to worry about that.)

There haven't been too many changes to the meta-updater-{raspberrypi,qemux86-64,etc} repos, and most of them are specific to u-boot, but there may be other things worth backporting in those as well.

leon-anavi commented:
Hi @patrickvacek,

thank you very much for the feedback. Based on your suggestions I have contributed several additional git commits. Please have a look at them.

Best regards, Leon

patrickvacek commented on recipes-sota/ostree/ostree_git.bb:
```diff
@@ -1,5 +1,6 @@
+LICENSE = "LGPLv2+"
 LIC_FILES_CHKSUM = "file://COPYING;md5=5f30f0716dfdd0d91eb439ebec522ec2"
 
 inherit autotools-brokensep pkgconfig systemd bash-completion gobject-introspection
```
Minor question: is `autotools-brokensep` still necessary here? It's no longer required for rocko and later versions.

OYTIS replied:
Are you sure? It looks like it's being used in rocko branch of poky in several places.

patrickvacek replied:
Our rocko version of this recipe doesn't use it, though, that's all.

leon-anavi replied:
It seems that replacing autotools-brokensep with autotools is OK for morty. I've rebased the ostree patch locally. Let me spin a couple of builds from scratch more to double check and I will provide further information.

Thanks, Leon
